datafusion/core/src/scheduler/task.rs

```diff
@@ -279,6 +280,7 @@ impl RecordBatchStream for ExecutionResultStream {
 
 /// The shared state of all [`Task`] created from the same [`PipelinePlan`]
 #[derive(Debug)]
+#[allow(dead_code)]
```

Comment:
What do you think about adding a function like

```rust
pub fn version(&self) -> &str {
...
}
```

Which would avoid the need for allow(dead_code) perhaps